woocommerce/woocommerce
# Ensure stock change notifications only fire when necessary

### Changes proposed in this Pull Request:

In #49583 we changed when the actions that trigger stock change notifications fire, to decouple them from order creation. However, it turns out what during product creation, the stock quantity update routines still run even when no stock is being set and even when the "manage stock" setting is set to false. Rather than try to untangle the web of why this happens (and potentially break other things in the process), this simply adds a check for the "manage stock" setting prior to firing the notif actions, and bail early if the product does not have managed stock.

It also adds a unit test, which is really more of an integration test, to ensure this behavior.

Fixes #50958

### How to test the changes in this Pull Request:

1. Install a plugin like [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/) on your test site so you can capture outgoing emails.
2. Create a new simple product. Give it a price, but do _not_ check the box to manage inventory. Publish.
3. Check the mail log and ensure that no email notification was sent about the new product being out of stock.
